tf2onnx/function/matrixbandpart.py

```diff
@@ -0,0 +1,90 @@
+
+def matrixbandpart_op(ctx, node, name, args):
+    # T output = matrix_band_part(T input, int num_lower, int num_upper)
+    # data-flow: first generate mask matrix and then use element-wise mul op
+    input_rank = len(ctx.get_shape(node.input[0]))
+    make_sure(input_rank == 2, error_msg="MatrixBandPart op: only rank2 is supported")
+    bandpart = [node.inputs[ind].get_tensor_value()[0] for ind in [1, 2]]
+    if bandpart not in [[-1, 0], [0, -1]]:
+        raise ValueError("MatrixBandPart op: only support Lower/Upper triangular part")
+
+    # methods to generate mask matrix: if lower triangular is needed, then generate column one by one
+    # otherwise row is generated one by one.
+    axis, counter_axis, squeeze_axis = (1, 0, 2) if bandpart == [-1, 0] else (0, 1, 1)
+    nodes = []
+    # 1: subgraph to implement tf.onelike(input[:, 0]),
+    # no need to worry about the dtype, because bool type is needed as Xor only support bool
+    node_name = utils.make_name("const_zero")
+    const_zero = ctx.make_const(name=node_name, np_val=np.array([0]).astype(np.int32))
+    first_col_or_row = ctx.make_node(op_type="Gather", inputs=[node.input[0], const_zero.output[0]],
+                                     attr={"axis": axis})
+    nodes.append(first_col_or_row)
+    first_col_or_row_casted = ctx.make_node(op_type="Cast", inputs=first_col_or_row.output,
+                                            attr={"to": onnx_pb.TensorProto.BOOL})
+    nodes.append(first_col_or_row_casted)
+    # line means one col or one row
+    zero_line = ctx.make_node(op_type="Xor", inputs=first_col_or_row_casted.output*2)
+    nodes.append(zero_line)
+    one_line = ctx.make_node(op_type="Not", inputs=zero_line.output)
+    nodes.append(one_line)
+
+    # 2: "loop" to generate mask matrix: generate col or row of matrix one by one
+    body_ins = [utils.make_onnx_inputs_outputs("trip", onnx_pb.TensorProto.INT64, []),
+                utils.make_onnx_inputs_outputs("cond", onnx_pb.TensorProto.BOOL, []),
+                utils.make_onnx_inputs_outputs("line", onnx_pb.TensorProto.BOOL, [-1, -1])]
+    body_outs = [utils.make_onnx_inputs_outputs("cond_out", onnx_pb.TensorProto.BOOL, []),
+                 utils.make_onnx_inputs_outputs("line_out", onnx_pb.TensorProto.BOOL, [-1, -1]),
+                 utils.make_onnx_inputs_outputs("res", onnx_pb.TensorProto.BOOL, [-1, -1])]
+    # body graph
+    node_name = utils.make_name("const_zero_bool")
+    const_zero_bool = ctx.make_const(name=node_name, np_val=np.array([[0]]).astype(np.bool))
+    slice_node = ctx.make_node(op_type="Slice", inputs=["line"],
+                               attr={"axes": [counter_axis], "starts": [0], "ends": [-1]})
+    new_line = ctx.make_node(op_type="Concat", inputs=[const_zero_bool.output[0], slice_node.output[0]],
+                             outputs=["line_out"], attr={"axis": counter_axis})
+    body_nodes = list([slice_node.op, new_line.op])
```

Comment:
why use list?

Reply:
to avoid warning in my local pycharm

Reply:
why?

Reply:
warning info is:   This list creation could be rewritten as a list literal less... (Ctrl+F1) 
Inspection info: This inspection detects situations when list creation could be rewritten with list literal